addons/maintenance/models/maintenance.py

```diff
@@ -190,17 +193,62 @@ def _read_group_category_ids(self, categories, domain, order):
         category_ids = categories._search([], order=order, access_rights_uid=SUPERUSER_ID)
         return categories.browse(category_ids)
 
+    def _create_new_request(self, date):
+        self.ensure_one()
+        self.env['maintenance.request'].create({
+                    'name': _('Preventive Maintenance - %s') % self.name,
+                    'request_date': fields.Date.to_string(date),
+                    'schedule_date': fields.Date.to_string(date),
+                    'category_id': self.category_id.id,
+                    'equipment_id': self.id,
+                    'maintenance_type': 'preventive',
+                    'owner_user_id': self.owner_user_id.id,
+                    'technician_user_id': self.technician_user_id.id,
+                    })
+
+    def _fill_up(self, from_date, to_date, to_end=False):
+        """
+            Will add maintenance requests between from_date and to_date with the equipment maintenance period as interval
+            Does nothing when the interval is too small
+            
+            to_end determines whether you need to create one at the end also or whether at the 
+            end there is a maintenance request already
+        """
+        self.ensure_one()
+        period = self.period
+        current_date = from_date + relativedelta(days=period)
+        while (current_date <= to_date):
+            if to_end or current_date + relativedelta(days=period) <= to_date:
+                self._create_new_request(current_date)
+            current_date = current_date + relativedelta(days=period)
+
     @api.model
     def _cron_generate_requests(self):
-        for equipment in self.search([]):
-            if equipment.period and equipment.next_action_date == date.today().strftime(DEFAULT_SERVER_DATE_FORMAT):
-                self.env['maintenance.request'].create({
-                    'name': _('Preventive Maintenance - %s' % equipment.next_action_date),
-                    'request_date': equipment.next_action_date,
-                    'category_id': equipment.category_id.id,
-                    'equipment_id': equipment.id,
-                    'maintenance_type': 'preventive',
-                })
+        show_until = fields.Date.from_string(fields.Datetime.now()) + relativedelta(months=1)
```

Comment:
Not sure tweaking on this will lead to better results: 
- The switch past/future is done because then it can do the same logic as between two existing maintenance requests
- it is done with > now, as the idea was to make extra where there is enough space in between, so the user can shift the dates...
- rrule -> not possible because of the > now, need to rethink